src/main/java/de/tum/in/www1/artemis/service/plagiarism/ProgrammingPlagiarismDetectionService.java

```diff
@@ -169,27 +169,27 @@ public File checkPlagiarismWithJPlagReport(long programmingExerciseId, float sim
     @NotNull
     private JPlagResult computeJPlagResult(ProgrammingExercise programmingExercise, float similarityThreshold, int minimumScore) {
         long programmingExerciseId = programmingExercise.getId();
-        final var targetPath = fileService.getUniquePathString(repoDownloadClonePath);
+        final var targetPath = fileService.getTemporaryUniquePath(repoDownloadClonePath, 60);
```

Comment:
IIRC it has been mentioned that the plagiarism checks for large courses at TUM can take very long. Is 60 minutes long enough so that the files are no longer in use before deletion starts?

Reply:
this was indeed something I wondered as well. 
@krusche do you know if 60 min are sufficient?

Reply:
yes, 60min is enough